src/vs/vscode.d.ts

```diff
@@ -5003,6 +5003,31 @@ declare module 'vscode' {
 		Workspace = 2
 	}
 
+	/**
+	 * Rerun behavior for a task.
+	 */
+	export enum RerunBehavior {
+		/**
+		 * Re-evaluate the variables associated with this task.
+		 */
+		reevaluate = 1,
+
+		/**
+		 * Use the variables associated with this task as they were evaluated on last run.
+		 */
+		useEvaluated = 2,
+	}
+
+	/**
+	 * Run options for a task.
+	 */
+	export interface RunOptions {
+		/**
+		 * Controls the behavior of a task when it is rerun.
+		 */
+		rerunBehavior?: RerunBehavior;
```

Comment:
what future run-option would there be? do have some in mind already?

Reply:
also, we say `rerunBehaviour` but it's all about evaluating or not. Is that the final list of rerun-options? are they all mutually exclusive like the two above?

We might be better of with a boolean in the `RunOptions`, depending on the default:

```ts
export interface RunOptions {
  reuseEvaluatedVariables?: boolean
}
```
or
```ts
export interface RunOptions {
  reevaluateVariables?: boolean
}
```

Reply:
I don't have other rerun behaviors in mind so I'm ok with making this a boolean. How about:

    export interface RunOptions {
        reevaluateOnRerun?: boolean
    }

I want to include that this only applies to rerun, since other task related properties are more general.

Reply:
Changed. Feel free to only look at vscode.d.ts!